src/igvfd/tests/test_upgrade_construct_library.py

```diff
@@ -17,3 +17,12 @@ def test_construct_library_upgrade_2_3(upgrader, construct_library_v2):
     assert value['schema_version'] == '3'
     assert 'origins' in value
     assert 'guide_library_details' in value
+
+
+def test_construct_library_upgrade_3_4(upgrader, construct_library_v3):
+    value = upgrader.upgrade(
+        'construct_library', construct_library_v3,
+        current_version='3', target_version='4')
+    assert value['schema_version'] == '4'
```

Comment:
please compare the content of publication_identifiers and refrences in value, it is not sufficient to check for presence if the content was lost